Keep track of to which instance secrets in plstore belong

hdurer opened the pull request:
While testing out [issue 149](https://github.com/jdenen/mastodon.el/issues/149) I had problems due to stale client information being cached.

With this change we store client information in the plstore keyed by the instance url and we also remember in the internal cache (the mastodon-client--client-details variable) for which instance we cached the data.

hdurer committed:
Keep track of to which instance secrets in plstore belong.

While testing out issue
149 (https://github.com/jdenen/mastodon.el/issues/149) I had problems
due to stale client information being cached.

With this change we store various pieces of information (the client
information in the plstore and the auth tokens) in alists keyed by the
instance url (and the plstore key contains the instance url as well to
allow us to store data per instance).

alexjgriffith commented on lisp/mastodon-client.el:
```diff
@@ -72,28 +73,30 @@
   (let ((plstore (plstore-open (mastodon-client--token-file)))
-        (client (mastodon-client--fetch)))
-    (plstore-put plstore "mastodon" client nil)
+         (client (mastodon-client--fetch)))
```
Indentation off by one char

Commit:
Explicity set output variables to "unlimited" when saving plstore.

Alex Griffith reported ellipses in the output which seems to indicate the printing was somehow truncated.
I cannot see anywhere in the code path for plstore-save that these variables be set explictly.

See https://www.gnu.org/software/emacs/manual/html_node/elisp/Output-Variables.html for an explanation of these variables.

alexjgriffith reviewed (dismissed):
LGTM
tested it out and looks PLSTORE is working as advertised and not leaving `...` in the place of `client_secret`

alexjgriffith reviewed (approved):
LGTM
